File: apps/mobile/lib/domain/entities/group.dart
class Group {
  final int id;
  final String name;
  final int ownerId;
  final DateTime createdAt;

  Group({
    required this.id,
    required this.name,
    required this.ownerId,
    required this.createdAt,
  });

  Group copyWith({
    int? id,
    String? name,
    int? ownerId,
    DateTime? createdAt,
  }) {
    return Group(
      id: id ?? this.id,
      name: name ?? this.name,
      ownerId: ownerId ?? this.ownerId,
      createdAt: createdAt ?? this.createdAt,
    );
  }
}

File: household-ledger-backend/src/domain/models/group.rs
use chrono::NaiveDateTime;
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Group {
    pub id: i64,
    pub name: String,
    pub owner_id: i64,
    #[sqlx(default)]
    pub created_at: NaiveDateTime,
}

